src/main/java/com/aws/greengrass/clientdevices/auth/PermissionEvaluationUtils.java

```diff
@@ -36,6 +37,7 @@ public final class PermissionEvaluationUtils {
     private static final Pattern SERVICE_RESOURCE_PATTERN = Pattern.compile(
             String.format(SERVICE_RESOURCE_FORMAT, SERVICE_PATTERN_STRING, SERVICE_RESOURCE_TYPE_PATTERN_STRING,
                     SERVICE_RESOURCE_NAME_PATTERN_STRING), Pattern.UNICODE_CHARACTER_CLASS);
+    private final WildcardTrie wildcardTrie = new WildcardTrie();
```

Comment:
why is this safe? Why can we reuse the same trie for multiple sessions (I'm assuming that this utils class is used across sessions)?

Reply:
I added a method reset the trie, this was to optimize not creating the children map multiple times.  But since replacing with hash map I could probably get away with instantiating the trie here, would need to test

Reply:
so hashmap in this use case is ~2x faster than concurrent hashmap? That's rough. Interesting.